js/modules/k6/experimental/fs/registry.go

```diff
@@ -0,0 +1,66 @@
+package fs
+
+import (
+	"io"
+	"path/filepath"
+	"sync"
+
+	"github.com/spf13/afero"
+)
+
+// fileRegistry is a registry of opened files.
+type fileRegistry struct {
+	// files holds a map of opened files, indexed by their path.
+	//
+	// The value is a pointer to the file's data. That way, we can cache the file's content and
+	// avoid re-reading it every time the file is opened. Importantly, this also means that if the
+	// file is modified from outside of k6, the changes will not be reflected in the file's data.
+	files map[string][]byte
+
+	// mutex is a mutex used to synchronize access to the files map.
+	//
+	// As we can have multiple VUs running in parallel, we need to make sure that the map is not
+	// accessed concurrently.
+	mutex sync.RWMutex
+}
+
+// newFileRegistry instantiates a new file registry.
+func newFileRegistry() *fileRegistry {
+	return &fileRegistry{
+		files: make(map[string][]byte),
+	}
+}
+
+// open opens the named file for reading.
+//
+// If the file was already opened, it returns a pointer to the cached file data. Otherwise, it
+// opens the file, reads its content, caches it, and returns a pointer to it.
+//
+// The file is always opened in read-only mode. The provided file path is cleaned before being
+// used.
+func (fr *fileRegistry) open(filename string, fromFs afero.Fs) ([]byte, error) {
```

Comment:
That's because we instantiate a single file registry module-wise, and we do that at the `RootModule` level, where we don't have the VU available to us yet, hence no access to the init environment yet: https://github.com/grafana/k6/blob/d6b43b9f16530f73241c510354f25951ba7be38a/js/modules/k6/experimental/fs/module.go#L39